File: 백준/Bronze/5566. 주사위 게임/주사위 게임.c
#include <stdio.h>

int main() {
    int N, M;
    scanf("%d %d", &N, &M);
    int board[N + 1];

    for (int i = 1; i <= N; i++) {
        scanf("%d", &board[i]);
    }

    int cur = 1;
    for (int j = 1; j <= M; j++) {
        int D;
        scanf("%d", &D);
        cur += D;
        if (cur >= N) {
            printf("%d\n", j);
            return 0;
        }
        cur += board[cur];
        if (cur >= N) {
            printf("%d\n", j);
            return 0;
        }
    }
    return 0;
}

File: 백준/Bronze/5566. 주사위 게임/주사위 게임.java
import java.util.Scanner;

public class Main {
    public static void main(String[] args) {
        Scanner sc = new Scanner(System.in);
        int N = sc.nextInt();
        int M = sc.nextInt();

        int[] board = new int[N + 1];

        for (int i = 1; i <= N; i++) {
            board[i] = sc.nextInt();
        }

        int cur = 1;
        for (int j = 1; j <= M; j++) {
            int D = sc.nextInt();
            cur += D;
            if (cur >= N) {
                System.out.println(j);
                return;
            }
            cur += board[cur];
            if (cur >= N) {
                System.out.println(j);
                return;
            }
        }
        sc.close();
    }
}